SimCore/src/SimCore/Generators/GenieGenerator.cxx

```diff
@@ -0,0 +1,395 @@
+  {
+    char* in_arr[3] = {const_cast<char*>(""),
+                       const_cast<char*>("--event-generator-list"),
+                       const_cast<char*>("EM")};
+    genie::RunOpt::Instance()->ReadFromCommandLine(3, in_arr);
+  }
+
+  // set message thresholds
+  genie::utils::app_init::MesgThresholds(message_threshold_file_);
+
+  // set tune info
+  genie::RunOpt::Instance()->SetTuneName(tune_);
+  if (!genie::RunOpt::Instance()->Tune()) {
+    EXCEPTION_RAISE("ConfigurationException", "No TuneId in RunOption.");
+  }
+  genie::RunOpt::Instance()->BuildTune();
+
+  // give it the splint file and require it
+  genie::utils::app_init::XSecTable(spline_file_, true);
+
+  // set GHEP print level (needed?)
+  genie::GHepRecord::SetPrintLevel(0);
+
+  // setup for event driver
+  evg_driver_.SetEventGeneratorList(
+      genie::RunOpt::Instance()->EventGeneratorList());
+  evg_driver_.SetUnphysEventMask(*genie::RunOpt::Instance()->UnphysEventMask());
+}
+
+void GenieGenerator::calculateTotalXS() {
+  // initializing...
+  xsec_total_ = 0;
+  ev_weighting_integral_.resize(targets_.size(), 0.0);
+
+  // calculate the total xsec per target...
+  for (size_t i_t = 0; i_t < targets_.size(); ++i_t) {
+    genie::InitialState initial_state(targets_[i_t], 11);
+    evg_driver_.Configure(initial_state);
+    evg_driver_.UseSplines();
+
+    // setup the initial election
+    TParticle initial_e;
+    initial_e.SetPdgCode(11);
+    auto elec_i_p = std::sqrt(energy_ * energy_ -
+                              initial_e.GetMass() * initial_e.GetMass());
+    initial_e.SetMomentum(elec_i_p * direction_[0], elec_i_p * direction_[1],
+                          elec_i_p * direction_[2], energy_);
+    TLorentzVector e_p4;
+    initial_e.Momentum(e_p4);
+
+    xsec_by_target_[i_t] = evg_driver_.XSecSum(e_p4);
+    xsec_total_ += xsec_by_target_[i_t] * abundances_[i_t];
+
+    ev_weighting_integral_[i_t] = xsec_total_;  // running sum
+
+    // print...
+    ldmx_log(debug) << "Target=" << targets_[i_t]
+                    << "\tAbundance=" << abundances_[i_t] << "\tXSEC="
+                    << xsec_by_target_[i_t] / genie::units::millibarn << "mb";
+  }
+  ldmx_log(debug) << "Total XSEC = " << xsec_total_ / genie::units::millibarn
+                  << " mb";
+
+  // renormalize our weighting integral
+  for (size_t i_t = 0; i_t < ev_weighting_integral_.size(); ++i_t)
+    ev_weighting_integral_[i_t] = ev_weighting_integral_[i_t] / xsec_total_;
+}
+
+GenieGenerator::GenieGenerator(const std::string& name,
+                               const framework::config::Parameters& p)
+    : PrimaryGenerator(name, p) {
+  fillConfig(p);
+
+  if (!validateConfig())
+    EXCEPTION_RAISE("ConfigurationException", "Configuration not valid.");
+
+  n_events_generated_ = 0;
+  initializeGENIE();
+  calculateTotalXS();
+}
+
+GenieGenerator::~GenieGenerator() {
+  std::cout << "--- GENIE Generation Summary BEGIN ---" << std::endl;
```

Comment:
I'm confused here: I thought these could become part of the logging system, is that not right?

Reply:
ok thanks!